src/db/z_db.erl

```diff
@@ -210,12 +218,14 @@ with_connection(F, Context) ->
             Result
         after
             return_connection(Connection, Context)
-	end.
+    end.
 
 
+-spec assoc_row(string(), #context{}) -> list(tuple()).
 assoc_row(Sql, Context) ->
     assoc_row(Sql, [], Context).
 
+-spec assoc_row(string(), parameters(), #context{}) -> proplists:proplist() | undefined.
```

Comment:
The spec in line 228 uses a record, please define a type for the record and use that instead.